sw/airborne/modules/decawave/decawave_serial_communication.c

```diff
@@ -0,0 +1,268 @@
+    _states[index].stateUpdated[j] = false;
+  }
+}
+
+/**
+ * Helper function that sets the booleans to false for all the remote drones (UWB_SERIAL_COMM_DIST_NUM_NODES)
+ */
+static void setAllNodeStatesFalse(void){
+  for (uint8_t i = 0; i < UWB_SERIAL_COMM_DIST_NUM_NODES; i++){
+    setNodeStatesFalse(i);
+  }
+}
+
+/**
+ * This function checks if all the states of all the distant nodes have at least once been updated.
+ * If all the states are updated, then do something with it! AKA CALLBACK TO MARIO
+ */
+static void checkStatesUpdated(void){
+  bool checkbool;
+  for (uint8_t i = 0; i < UWB_SERIAL_COMM_DIST_NUM_NODES; i++){
+    checkbool = true;
+    for (uint8_t j = 0; j < UWB_SERIAL_COMM_NODE_STATE_SIZE; j++){
+      checkbool = checkbool && _states[i].stateUpdated[j];
+    }
+    if (checkbool){
+      AbiSendMsgUWB_COMMUNICATION(UWB_COMM_ID, i, _states[i].r, _states[i].vx, _states[i].vy, _states[i].z);
+      setNodeStatesFalse(i);
+    }
+  }
+
+}
+
+/**
+ * Function for receiving serial data.
+ * Only receives serial data that is between the start and end markers. Discards all other data.
+ * Stores the received data in _tempBuffer, and after decodes the high bytes and copies the final
+ * message to the corresponding message in _messages.
+ */
+void getSerialData(void){
+  while (SerialChAvailable()){
+    _varByte = SerialGetch();
+    if (_varByte == UWB_SERIAL_COMM_START_MARKER){
+      _bytesRecvd = 0;
+      _inProgress = true;
+    }
+
+    if (_inProgress){
+      _tempBuffer[_bytesRecvd] = _varByte;
+      _bytesRecvd++;
+    }
+
+    if (_varByte == UWB_SERIAL_COMM_END_MARKER){
+      _inProgress = false;
+      _allReceived = true;
+
+      decodeHighBytes();
+    }
+  }
+}
+
+/**
+ * Function for decoding the high bytes of received serial data and saving the message.
+ * Since the start and end marker could also be regular payload bytes (since they are simply the values
+ * 254 and 255, which could also be payload data) the payload values 254 and 255 have been encoded
+ * as byte pairs 253 1 and 253 2 respectively. Value 253 itself is encoded as 253 0.
+ *  This function will decode these back into values the original payload values.
+ */
+static void decodeHighBytes(void){
+  _dataRecvCount = 0;
+  float tempfloat;
+  uint8_t thisAddress = _tempBuffer[1];
+  uint8_t msgFrom = _tempBuffer[2];
+  uint8_t msgType = _tempBuffer[3];
+  uint8_t nodeIndex = msgFrom -1 - (uint8_t)(thisAddress<msgFrom);
+  for (uint8_t i = 4; i<_bytesRecvd-1; i++){ // Skip the begin marker (0), this address (1), remote address (2), message type (3), and end marker (_bytesRecvd-1)
+    _varByte = _tempBuffer[i];
+    if (_varByte == UWB_SERIAL_COMM_SPECIAL_BYTE){
+      i++;
+      _varByte = _varByte + _tempBuffer[i];
+    }
+    if(_dataRecvCount<=UWB_SERIAL_COMM_FLOAT_SIZE){
+      _recvBuffer[_dataRecvCount] = _varByte;
+    }
+    _dataRecvCount++;
+  }
+  if(_dataRecvCount==UWB_SERIAL_COMM_FLOAT_SIZE){
```

Comment:
when you're done, don't forget to run the `fix_code_style.sh` script on your files